app/components/projects/row_component.rb

```diff
@@ -112,8 +119,26 @@ def project_status
     def status_explanation
       return nil unless user_can_view_project?
 
-      if project.status_explanation
-        content_tag :div, helpers.format_text(project.status_explanation), class: 'wiki'
+      if project.status_explanation.present? && project.status_explanation
+        concat content_tag :p, Nokogiri::HTML(project.status_explanation).text, class: 'project-long-text', id: "#{project.id}-status-explanation"
+        render(Primer::Alpha::Dialog.new(id: "dialog-#{project.id}-status-explanation", title: I18n.t('activerecord.attributes.project.status_explanation'))) do |component|
+          component.with_show_button(scheme: :link, display: (includes_table(project.status_explanation) ? :block : :none)) { I18n.t('js.label_expand') }
+          component.with_body { helpers.format_text(project.status_explanation) }
+          component.with_header(show_divider: true)
+        end
+      end
+    end
+
+    def description
+      return nil unless user_can_view_project?
+
+      if project.description.present?
+        concat content_tag :p, Nokogiri::HTML(project.description).text, class: 'project-long-text', id: "#{project.id}-description"
+        render(Primer::Alpha::Dialog.new(id: "dialog-#{project.id}-description", title: I18n.t('activerecord.attributes.project.description'))) do |component|
+          component.with_show_button(scheme: :link, display: (includes_table(project.description) ? :block : :none)) { I18n.t('js.label_expand') }
+          component.with_body { helpers.format_text(project.description) }
+          component.with_header(show_divider: true)
+        end
```

Comment:
The code for displaying the full text in a dialog is repeated three times in this component. Given that the same functionality will be required for displaying the project attributes on the overview page as well, let's extract the functionality into a component that can be reused. This doesn't even have to be project specific but rather a component able to display the full text of a single attribute.